### PR TITLE
Docs: convert boards to cmake

### PR DESCRIPTION
**Pull request by mbolivar**
Convert the ARM boards' usage of "make" commands to cmake using zephyr-app-commands.

With x86 done, this leaves ARC, NIOS II, RISC V 32, and XTensa. That's actually not very much work, since those arches don't have very many boards. That will come next, and boards/ will be converted.

Note that there are a few cases where "make" commands were deliberately left when building other tools.

### CODE REVIEW

File: boards/arm/frdm_k64f/doc/frdm_k64f.rst [hunk 1]
```diff
@@ -198,29 +198,30 @@ communication over USB.
...
+:ref:`nxp_opensda_pyocd` tools.
 
-   $ cd <zephyr_root_path>
-   $ . zephyr-env.sh
-   $ cd samples/hello_world/
-   $ make BOARD=frdm_k64f flash
+.. zephyr-app-commands::
+   :zephyr-app: samples/hello_world
+   :board: frdm_k64f
+   :goals: build flash
```
Comment (carlescufi) [inline boards/arm/frdm_k64f/doc/frdm_k64f.rst]: should be :goals: flash if you want to replicate the original?

File: boards/arm/frdm_kl25z/doc/frdm_kl25z.rst [hunk 2]
```diff
@@ -126,29 +126,29 @@ communication over USB.
...
+:ref:`nxp_opensda_pyocd` tools.
 
-   $ cd <zephyr_root_path>
-   $ . zephyr-env.sh
-   $ cd samples/hello_world/
-   $ make BOARD=frdm_kl25z flash
+.. zephyr-app-commands::
+   :zephyr-app: samples/hello_world
+   :board: frdm_kl25z
+   :goals: build flash
```
Comment (carlescufi) [inline boards/arm/frdm_kl25z/doc/frdm_kl25z.rst]: should be :goals: flash if you want to replicate the original?

File: boards/arm/hexiwear_k64/doc/hexiwear_k64.rst [hunk 3]
```diff
@@ -198,16 +203,12 @@ Flashing
...
+:ref:`nxp_opensda_pyocd` tools.
 
-   $ cd <zephyr_root_path>
-   $ . zephyr-env.sh
-   $ cd samples/hello_world/
-   $ make BOARD=hexiwear_k64 flash
+.. zephyr-app-commands::
+   :zephyr-app: samples/hello_world
+   :board: hexiwear_k64
+   :goals: build flash
```
Comment (carlescufi) [inline boards/arm/hexiwear_k64/doc/hexiwear_k64.rst]: should be :goals: flash if you want to replicate the original?

File: boards/arm/qemu_cortex_m3/doc/board.rst [hunk 4]
```diff
@@ -71,9 +71,10 @@ Programming and Debugging
 Use this configuration to run basic Zephyr applications and kernel tests in the QEMU
 emulated environment, for example, with the :ref:`synchronization_sample`:
 
-.. code-block:: console
-
-        $ make -C samples/synchronization BOARD=qemu_cortex_m3 run
+.. zephyr-app-commands::
+   :zephyr-app: samples/synchronization
+   :board: qemu_cortex_m3
+   :goals: build run
```
Comment (carlescufi) [inline boards/arm/qemu_cortex_m3/doc/board.rst]: run only

**Review by carlescufi — APPROVED** [review]
This is thorough and exhaustive, great work!. I have a few comments but they are minor.

File: boards/arm/frdm_k64f/doc/frdm_k64f.rst (hunk 1 above)
Comment (mbolivar) [reply boards/arm/frdm_k64f/doc/frdm_k64f.rst]: Yes, I did these separately on purpose as I think it is good practice to  build first before running, but you're right that I shouldn't be doing that in this series. I'll revise all this.

**Comment by mbolivar** [comment]
@carlescufi thanks for the review! Comments addressed.

File: boards/arm/stm32f3_disco/doc/stm32f3_disco.rst [hunk 5]
```diff
@@ -153,33 +153,24 @@ Flashing
...
-The sample application :ref:`hello_world` is being used in this tutorial:
-
-.. code-block:: console
-
-   $<zephyr_root_path>/samples/hello_world
-
-To build the Zephyr kernel and application, enter:
-
-.. code-block:: console
+First, connect the STM32F469I-DISCO Discovery kit to your host computer using
```
Comment (mbolivar) [inline boards/arm/stm32f3_disco/doc/stm32f3_disco.rst]: Wrong board, should be fixed

File: boards/arm/stm32f3_disco/doc/stm32f3_disco.rst [hunk 6]
```diff
@@ -95,7 +95,7 @@ Other hardware features are not yet supported on Zephyr porting.
 
 The default configuration can be found in the defconfig file:
 
-	``boards/arm/stm32f3_disco/stm32f3_disco_defconfig``
+        ``boards/arm/stm32f3_disco/stm32f3_disco_defconfig``
```
Comment (mbolivar) [inline boards/arm/stm32f3_disco/doc/stm32f3_disco.rst]: Unrelated change

File: boards/arm/nucleo_l432kc/doc/nucleol432kc.rst [hunk 7]
```diff
@@ -168,50 +168,36 @@ Default settings are 115200 8N1.
...
-.. code-block:: console
-
-   $ cd <openocd-stm32_path>
-   $ stm32_flsh l4 $ZEPHYR_BASE/samples/hello_world/outdir/nucleo_l432kc/zephyr.bin
-
-Run a serial host program to connect with your Nucleo board.
+   $ minicom -D /dev/ttyACM0
 
-.. code-block:: console
+Now build and flash an applicaiton. Here is an example for
```
Comment (mbolivar) [inline boards/arm/nucleo_l432kc/doc/nucleol432kc.rst]: typo

File: boards/arm/stm32f429i_disc1/doc/stm32f429i_disc1.rst [hunk 8]
```diff
@@ -26,10 +26,10 @@ some highlights of the STM32F429I-DISC1 board:
 - L3GD20, ST-MEMS motion sensor 3-axis digital output gyroscope
 - Six LEDs
 
-	- LD1 (red/green) for USB communication
-	- LD2 (red) for 3.3 V power-on
-	- Two user LEDs: LD3 (green), LD4 (red)
-	- Two USB OTG LEDs: LD5 (green) VBUS and LD6 (red) OC (over-current)
+        - LD1 (red/green) for USB communication
```
Comment (mbolivar) [inline boards/arm/stm32f429i_disc1/doc/stm32f429i_disc1.rst]: unrelated changes

File: boards/arm/stm32f429i_disc1/doc/stm32f429i_disc1.rst [hunk 9]
```diff
@@ -98,7 +98,7 @@ Other hardware features are not yet supported on Zephyr porting.
 
 The default configuration can be found in the defconfig file:
 
-	``boards/arm/stm32f429i_disc1/stm32f429i_disc1_defconfig``
+        ``boards/arm/stm32f429i_disc1/stm32f429i_disc1_defconfig``
```
Comment (mbolivar) [inline boards/arm/stm32f429i_disc1/doc/stm32f429i_disc1.rst]: unrelated

File: boards/arm/stm32f429i_disc1/doc/stm32f429i_disc1.rst [hunk 10]
```diff
@@ -136,6 +136,10 @@ The default communication settings are 115200 8N1.
 Programming and Debugging
 *************************
 
+Applications for the ``stm32f429i_disc1.rst`` board configuration can be built
```
Comment (mbolivar) [inline boards/arm/stm32f429i_disc1/doc/stm32f429i_disc1.rst]: delete .rst

File: boards/arm/stm32f469i_disco/doc/stm32f469i_disco.rst [hunk 11]
```diff
@@ -102,7 +102,7 @@ Other hardware features are not yet supported on Zephyr porting.
 
 The default configuration can be found in the defconfig file:
 
-	``boards/arm/stm32f469i_disco/stm32f469i_disco_defconfig``
+        ``boards/arm/stm32f469i_disco/stm32f469i_disco_defconfig``
```
Comment (mbolivar) [inline boards/arm/stm32f469i_disco/doc/stm32f469i_disco.rst]: unrelated

File: boards/arm/stm32l496g_disco/doc/stm32l496g_disco.rst [hunk 12]
```diff
@@ -189,62 +189,50 @@ Programming and Debugging
...
-needed to flash and debug STM32 development boards.
-
-.. code-block:: console
-
-   $ git clone  https://github.com/erwango/openocd-stm32.git
-
-Then follow instructions in README.md
+STM32L496G Discovery board includes an ST-LINK/V2-1 embedded debug
+tool interface.  This interface is supported by openocd version
+v0.10.0, which has been available since Zephyr SDK 0.9.2.
```
Comment (mbolivar) [inline boards/arm/stm32l496g_disco/doc/stm32l496g_disco.rst]: v0.9.2

**Review by mbolivar — COMMENTED** [review]
Caught a few of my own issues, will push a fix momentarily

**Comment by mbolivar** [comment]
I had some time to get the rest of the boards today.

File: boards/arm/nucleo_f103rb/doc/nucleof103rb.rst [hunk 13]
```diff
@@ -124,33 +128,26 @@ This interface is supported by the openocd version included in the Zephyr SDK.
...
-
-.. code-block:: console
-
-   $ make BOARD=nucleo_f103rb debug
+You can debug an application in the usual way.  Here is an example for the
+:ref:`blinky-sample` application.
 
+.. zephyr-app-commands::
+   :zephyr-app: samples/basic-blinky
+   :board: nucleof103rb
```
Comment (nashif) [inline boards/arm/nucleo_f103rb/doc/nucleof103rb.rst]: nucleo_f103rb?

File: boards/arm/nucleo_f103rb/doc/nucleof103rb.rst [hunk 14]
```diff
@@ -124,33 +128,26 @@ This interface is supported by the openocd version included in the Zephyr SDK.
...
-Connect the Nucleo F103RB to your host computer using the USB port.
-Then, enter the following command:
+Here is an example for the :ref:`blinky-sample` application.
 
-.. code-block:: console
-
-   $ make BOARD=nucleo_f103rb flash
+.. zephyr-app-commands::
+   :zephyr-app: samples/basic/blinky
+   :board: nucleof103rb
```
Comment (nashif) [inline boards/arm/nucleo_f103rb/doc/nucleof103rb.rst]: nucleo_f103rb

File: boards/arm/nucleo_f334r8/doc/nucleof334r8.rst [hunk 15]
```diff
@@ -128,45 +132,28 @@ This interface is supported by the openocd version included in Zephyr SDK.
...
-   $ make BOARD=nucleo_f334r8 flash
-
-Run a serial host program to connect with your Nucleo board:
-
-.. code-block:: console
-
-   $ minicom -D /dev/ttyACM0
+.. zephyr-app-commands::
+   :zephyr-app: samples/basic/blinky
+   :board: nucleof334r8
```
Comment (nashif) [inline boards/arm/nucleo_f334r8/doc/nucleof334r8.rst]: nucleo_f103rb, and many others below

**Review by nashif — APPROVED** [review]

**Review by nashif — CHANGES_REQUESTED** [review]
duh, wanted something else...

File: boards/arm/nucleo_f103rb/doc/nucleof103rb.rst (hunk 14 above)
Comment (mbolivar) [reply boards/arm/nucleo_f103rb/doc/nucleof103rb.rst]: Thanks for catching that. I was just copying the file name in as I assumed that was the convention after they stopped just being called "board.rst"

File: boards/arm/nucleo_f334r8/doc/nucleof334r8.rst (hunk 15 above)
Comment (mbolivar) [reply boards/arm/nucleo_f334r8/doc/nucleof334r8.rst]: Seems you've copy pasted as well ;)

**Comment by mbolivar** [comment]
@nashif nice catch! fixed.

**Review by nashif — APPROVED** [review]
